test/TestUtils.ts

```diff
@@ -0,0 +1,426 @@
+import childProcess, { ChildProcessWithoutNullStreams } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+import { BigNumberish, constants, utils, Wallet } from 'ethers';
+import chaiAsPromised from 'chai-as-promised';
+import { expect, use } from 'chai';
+import config from 'config';
+
+import {
+  RelayHub,
+  SmartWalletFactory,
+  CustomSmartWalletFactory,
+  IForwarder,
+  RelayHub__factory,
+} from '@rsksmart/rif-relay-contracts';
+import {
+  AppConfig,
+  defaultEnvironment,
+  RelayHubConfiguration,
+  ServerConfigParams,
+  sleep,
+} from '@rsksmart/rif-relay-server';
+import {
+  DeployRequest,
+  deployRequestType,
+  EnvelopingRequest,
+  getEnvelopingRequestDataV4Field,
+  HttpClient,
+  HttpWrapper,
+  HubInfo,
+  isDeployRequest,
+  RelayRequest,
+  relayRequestType,
+} from '@rsksmart/rif-relay-client';
+import { ethers } from 'hardhat';
+import { _TypedDataEncoder } from 'ethers/lib/utils';
+
+use(chaiAsPromised);
+
+const SERVER_WORK_DIR = '/tmp/enveloping/test/server';
+const ONE_FIELD_IN_BYTES = 32;
+
+type StartRelayParams = {
+  serverConfig: ServerConfigParams;
+  delay?: number;
+  stake?: BigNumberish;
+  relayOwner: string;
+  relayHubAddress: string;
+};
+
+const provider = ethers.provider;
+
+const startRelay = async (options: StartRelayParams) => {
+  const { serverConfig, delay, stake, relayOwner, relayHubAddress } = options;
+
+  fs.rmSync(SERVER_WORK_DIR, { recursive: true, force: true });
+
+  const originalConfig = { ...config };
+
+  config.util.extendDeep(originalConfig, {
+    ...serverConfig,
+    contracts: {
+      relayHubAddress,
+    },
+    app: {
+      workdir: SERVER_WORK_DIR,
+      devMode: true,
+      checkInterval: 10,
+      logLevel: 5,
+    },
+  });
+
+  const url = buildServerUrl();
+
+  const runServerPath = path.resolve(
+    __dirname,
+    '../node_modules/@rsksmart/rif-relay-server/dist/commands/Start.js'
+  );
+
+  const proc: ChildProcessWithoutNullStreams & { alreadyStarted?: number } =
+    childProcess.spawn('node', [runServerPath]);
+
+  const { relayManagerAddress } = await waitForFunction(
+    getServerStatus,
+    url,
+    "can't ping server",
+    3
+  );
+
+  console.log('Relay Server Manager Address', relayManagerAddress);
+
+  await provider.getSigner(relayOwner).sendTransaction({
+    to: relayManagerAddress,
+    value: utils.parseEther('2'),
+  });
+
+  const relayHub = RelayHub__factory.connect(relayHubAddress, provider);
+
+  await relayHub.stakeForAddress(relayManagerAddress, delay || 2000, {
+    from: relayOwner,
+    value: stake || utils.parseEther('1'),
+  });
+
+  const { relayWorkerAddress } = await waitForFunction(
+    getServerReady,
+    url,
+    'timed out waiting for relay to get staked and registered',
+    25,
+    500
```

Comment:
why 25 and 500?

Reply:
Ill put it as a constant, but it was the number previously set.